### PR TITLE
samples: usb: Update run instructions for testusb sample

### LINKED ISSUE #61971: Missing documentation in testusb README
**zephyr/samples/subsys/usb/testusb/README.rst** does not mention that you should unplug the device and plug it again after  command:  ‘sudo sh -c "echo 0x2fe3 0x0009 0 0x0525 0xa4a0 > /sys/bus/usb/drivers/usbtest/new_id"’
If not done full test suite is not available. Please update the documentation.

### PR DESCRIPTION
**Pull request by ro4i7**
samples: usb: Update run instructions for testusb sample

Fixes: #61971

This commit updates the README documentation for the `testusb` Zephyr 
sample to provide clearer instructions to users. Specifically, it 
adds a note to remind users to unplug and plug in the USB device
after running the command to load the USB driver ID. Without 
this step, the full test suite is not available. The updated 
README now includes this important information to 
improve the user experience and prevent potential issues.

Signed-off-by: Rohit Raj <loginrohitraj@gmail.com>

### CODE REVIEW

**Comment by ro4i7** [comment]
Hello @henrikbrixandersen  ,please check this and proceed

File: samples/subsys/usb/testusb/README.rst [hunk 1]
```diff
@@ -49,18 +49,19 @@ To run USB tests:
...
-   Since we use the "Gadget Zero" interface we specify reference device
-   ``0525:a4a0``.
+   Since we use the "Gadget Zero" interface, specify the reference device ``0525:a4a0``.
 
    .. code-block:: console
 
       $ sudo sh -c "echo 0x2fe3 0x0009 0 0x0525 0xa4a0 > /sys/bus/usb/drivers/usbtest/new_id"
 
-#. Use the ``testusb`` tool in ``linux/tools/usb`` inside Linux kernel source directory
+   **Note:** After running the above command, make sure to unplug the USB device and plug it back in for the full test suite to be available.
```
Comment (kartben) [inline samples/subsys/usb/testusb/README.rst:62]:
```suggestion
   .. note: After running the above command, make sure to unplug the USB device and plug it back in for the full test suite to be available.
```

**Review by kartben — CHANGES_REQUESTED** [review]
Thanks! I've commented with a minor nit to make the note more eye-catching.
Please check https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-guidelines and update amend commit message / sign off accordingly before (force) pushing again to your branch.

Thanks again for your first PR to the project :)

**Comment by ro4i7** [comment]
> Hello @henrikbrixandersen ,please check this and proceed

Hello @kartben , please check it i have updated with your suggestions. Thank you

**Review by tmon-nordic — CHANGES_REQUESTED** [review]
You should squash the commits into one and make commit message less cryptic.

File: samples/subsys/usb/testusb/README.rst [hunk 2]
```diff
@@ -49,18 +49,19 @@ To run USB tests:
...
-   Since we use the "Gadget Zero" interface we specify reference device
-   ``0525:a4a0``.
+   Since we use the "Gadget Zero" interface, specify the reference device ``0525:a4a0``.
 
    .. code-block:: console
 
       $ sudo sh -c "echo 0x2fe3 0x0009 0 0x0525 0xa4a0 > /sys/bus/usb/drivers/usbtest/new_id"
 
-#. Use the ``testusb`` tool in ``linux/tools/usb`` inside Linux kernel source directory
+   .. note: After running the above command, make sure to unplug the USB device and plug it back in for the full test suite to be available.
```
Comment (jfischer-no) [inline samples/subsys/usb/testusb/README.rst:62]: Line length not acceptable.

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: samples/subsys/usb/testusb/README.rst (hunk 2 above)
Comment (henrikbrixandersen) [reply samples/subsys/usb/testusb/README.rst:62]: Please keep the line length below 100 characters as per https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style

File: samples/subsys/usb/testusb/README.rst [hunk 3]
```diff
@@ -49,18 +49,19 @@ To run USB tests:
...
-   Since we use the "Gadget Zero" interface we specify reference device
-   ``0525:a4a0``.
+   Since we use the "Gadget Zero" interface, specify the reference device ``0525:a4a0``.
 
    .. code-block:: console
 
       $ sudo sh -c "echo 0x2fe3 0x0009 0 0x0525 0xa4a0 > /sys/bus/usb/drivers/usbtest/new_id"
 
-#. Use the ``testusb`` tool in ``linux/tools/usb`` inside Linux kernel source directory
+   .. note: After running the command, unplug and re-plug the USB device to enable the full test suite.
```
Comment (kartben) [inline samples/subsys/usb/testusb/README.rst:62]: apologies, I made a typo earlier :( 

```suggestion
   .. note:: After running the command, unplug and re-plug the USB device to enable the full test suite.
```

**Review by kartben — CHANGES_REQUESTED** [review]
OK, almost there, thanks for your patience! 
Please do squash all your commits into just one, and make sure you do the following as per the contribution guidelines:
- Proper title (ex. "samples: usb: Update run instructions for testusb sample")
- Descriptive message (careful with line length there too)
- Signed off commit

You can then force push the single, squashed, commit, to your branch.

**Comment by kartben** [comment]
@ro4i7 I see that there seems to be some confusion as you are updating the PR description, but what matters is the actual commit, since that's what ends up in the Git repo for everyone to see, independently of Github. The description and title you put in the PR are great, so if you can use them for the squashed commit, that will be perfect.

`git rebase origin/main --interactive` --> you can then squash all commits into the same one, and reword the commit message

```
reword 2a3dbf6eaf Update README.rst
fixup 72aa435e6d Update samples/subsys/usb/testusb/README.rst
fixup 2c26c478ab Update README.rst
fixup 738241a2f3 Update README.rst
```

This will then prompt you to *reword* the first commit (that's where you use the title/description/sign off that you have put in the initial comment of this PR), and then the 3 other commits will end up squashed into the first, with their "useless" commit message just being discarded. Hope this helps!

**Comment by kartben** [comment]
@ro4i7 let me know if you need help with regards to my latest comment. Thank you!

**Comment by kartben** [comment]
also note that you still have a line that's over 100 characters in there so would be great if you could also fix that. Thanks!

**Comment by kartben** [comment]
@ro4i7 will you be able to revisit this PR and address pending comments? Thanks!

**Pull request closed without merging**